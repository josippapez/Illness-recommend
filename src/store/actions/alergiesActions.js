import axios from 'axios';
import { ALERGIES_LIST_FETCHED, SET_ERROR_ALERGIES_LIST } from '../types';

export const getAllAlergies = () => {
  return (dispatch, getState) => {
    axios({
      method: 'GET',
      url: `${process.env.REACT_APP_API_URL}/alergies`,
      withCredentials: true,
    })
      .then(response => {
        dispatch(setAlergiesList(response.data));
      })
      .catch(error =>
        dispatch({
          type: SET_ERROR_ALERGIES_LIST,
          payload: {
            error: error.message,
            status: error.status,
          },
        })
      );
  };
};

export const setAlergiesList = data => ({
  type: ALERGIES_LIST_FETCHED,
  payload: data,
});
