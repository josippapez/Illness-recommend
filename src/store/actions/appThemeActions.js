import { SET_THEME } from '../types';

export const setDarkTheme = data => ({
  type: SET_THEME,
  payload: data,
});
