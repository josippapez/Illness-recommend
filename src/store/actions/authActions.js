import axios from 'axios';
import { SET_ERROR_USER, USER_LOGGED_IN, USER_LOGGED_OUT } from '../types';

export const registerUser = (name, email, password) => {
  return (dispatch, getState) => {
    axios({
      method: 'POST',
      url: `${process.env.REACT_APP_API_URL}/authentication/register`,
      data: {
        name,
        email,
        password,
      },
    })
      .then(response => {
        dispatch(loginUser(email, password));
      })
      .catch(error =>
        dispatch(
          loginError({
            message: error.response.data.message,
            status: error.response.status,
          })
        )
      );
  };
};

export const loginUser = (email, password) => {
  return (dispatch, getState) => {
    axios({
      method: 'POST',
      url: `${process.env.REACT_APP_API_URL}/authentication/log-in`,
      withCredentials: true,
      data: {
        email,
        password,
      },
    })
      .then(response => {
        dispatch(userLoggedIn(response.data));
      })
      .catch(error =>
        dispatch(
          loginError({
            message: error.response.data.message,
            status: error.response.status,
          })
        )
      );
  };
};

export const userLoggedIn = user => ({
  type: USER_LOGGED_IN,
  payload: {
    id: user.id,
    email: user.email,
    name: user.name,
    role: user.role
  },
});

export const userLoggedOut = () => ({
  type: USER_LOGGED_OUT,
});

export const loginError = error => ({
  type: SET_ERROR_USER,
  payload: {
    error: error.message,
    status: error.status,
  },
});
