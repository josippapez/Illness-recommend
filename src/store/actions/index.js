export * from './authActions';
export * from './usersActions';
export * from './alergiesActions';
export * from './symptomsActions';
export * from './medicationActions';
export * from './patientActions';
