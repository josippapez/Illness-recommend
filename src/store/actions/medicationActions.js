import axios from 'axios';
import Cookies from 'js-cookie';
import {
  MEDICATION_INFO_FETCHED,
  MEDICATION_INFO_FETCHED_ERROR,
  MEDICATION_LIST_FETCHED,
  SET_ERROR_MEDICATION_LIST,
} from '../types';

export const getAllMedications = () => {
  return (dispatch, getState) => {
    axios({
      method: 'GET',
      url: `${process.env.REACT_APP_API_URL}/medications/`,
      headers: {
        Authorization: `Bearer ${Cookies.get('Accesstoken')}`,
      },
    })
      .then(response => {
        dispatch(setMedicationsList(response.data));
      })
      .catch(error =>
        dispatch({
          type: SET_ERROR_MEDICATION_LIST,
          payload: {
            message: error.message,
            error: error.message,
            status: error.status,
          },
        })
      );
  };
};

export const searchMedicationsByText = query => {
  return (dispatch, getState) => {
    axios({
      method: 'GET',
      url: `${process.env.REACT_APP_API_URL}/medications/search?search=${query}`,
      headers: {
        Authorization: `Bearer ${Cookies.get('Accesstoken')}`,
      },
    })
      .then(response => {
        dispatch(setMedicationsList(response.data));
      })
      .catch(error =>
        dispatch({
          type: SET_ERROR_MEDICATION_LIST,
          payload: {
            message: error.message,
            error: error.message,
            status: error.status,
          },
        })
      );
  };
};

export const getMedicationsBySymptomsAndAlergies = symptoms => {
  return (dispatch, getState) => {
    axios({
      method: 'POST',
      url: `${process.env.REACT_APP_API_URL}/medications/find-by-symptoms`,
      headers: {
        Authorization: `Bearer ${Cookies.get('Accesstoken')}`,
      },
      data: symptoms,
    })
      .then(response => {
        dispatch(
          setErrorMedicationInfo({
            message: response.data.successMessage,
            error: null,
            status: response.status,
          })
        );
        dispatch(
          setErrorMedicationInfo({ message: null, error: null, status: null })
        );
        dispatch(setMedicationsList(response.data));
      })
      .catch(error =>
        dispatch({
          type: SET_ERROR_MEDICATION_LIST,
          payload: {
            message: error.message,
            error: error.message,
            status: error.status,
          },
        })
      );
  };
};

export const createMedication = medication => {
  return (dispatch, getState) => {
    axios({
      method: 'POST',
      url: `${process.env.REACT_APP_API_URL}/medications/`,
      headers: {
        Authorization: `Bearer ${Cookies.get('Accesstoken')}`,
      },
      data: medication,
    })
      .then(response => {
        dispatch(
          setErrorMedicationInfo({
            message: response.data.successMessage,
            error: null,
            status: response.status,
          })
        );
        dispatch(
          setErrorMedicationInfo({ message: null, error: null, status: null })
        );
        dispatch(getAllMedications());
      })
      .catch(error => {
        dispatch(
          setErrorMedicationInfo({
            error: error.response.data,
            status: error.status,
          })
        );
      });
  };
};

export const updateMedication = medication => {
  return (dispatch, getState) => {
    axios({
      method: 'PATCH',
      url: `${process.env.REACT_APP_API_URL}/medications/`,
      headers: {
        Authorization: `Bearer ${Cookies.get('Accesstoken')}`,
      },
      data: medication,
    })
      .then(response => {
        dispatch(
          setErrorMedicationInfo({
            message: response.data.successMessage,
            error: null,
            status: response.status,
          })
        );
        dispatch(
          setErrorMedicationInfo({ message: null, error: null, status: null })
        );
        dispatch(getAllMedications());
      })
      .catch(error =>
        dispatch(
          setErrorMedicationInfo({
            error: error.response.data,
            status: error.status,
          })
        )
      );
  };
};

export const removeMedicationById = id => {
  return (dispatch, getState) => {
    axios({
      method: 'DELETE',
      url: `${process.env.REACT_APP_API_URL}/medications/delete`,
      headers: {
        Authorization: `Bearer ${Cookies.get('Accesstoken')}`,
      },
      data: { id },
    })
      .then(response => {
        dispatch(
          setErrorMedicationInfo({
            message: response.data.successMessage,
            error: null,
            status: response.status,
          })
        );
        dispatch(
          setErrorMedicationInfo({ message: null, error: null, status: null })
        );
        dispatch(getAllMedications());
      })
      .catch(error =>
        dispatch(
          setErrorMedicationInfo({
            error: error.response.data,
            status: error.status,
          })
        )
      );
  };
};

export const setMedicationsList = data => ({
  type: MEDICATION_LIST_FETCHED,
  payload: data,
});

export const medicationInfoFetched = data => ({
  type: MEDICATION_INFO_FETCHED,
  payload: data,
});

export const setErrorMedicationInfo = data => ({
  type: MEDICATION_INFO_FETCHED_ERROR,
  payload: data,
});
