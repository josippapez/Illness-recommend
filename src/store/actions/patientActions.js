import axios from 'axios';
import Cookies from 'js-cookie';
import {
  PATIENT_INFO_FETCHED,
  PATIENT_INFO_FETCHED_ERROR,
  PATIENT_LIST_FETCHED,
} from '../types';

export const getAllPatientsForAdmin = () => {
  return (dispatch, getState) => {
    axios({
      method: 'GET',
      url: `${process.env.REACT_APP_API_URL}/patients-details/`,
      headers: {
        Authorization: `Bearer ${Cookies.get('Accesstoken')}`,
      },
    })
      .then(response => {
        dispatch(setPatientsList(response.data));
      })
      .catch(error =>
        setErrorCurrentPatientInfo({
          message: error.message,
          error: error.message,
          status: error.status,
        })
      );
  };
};

export const getAllPatientsForUser = () => {
  return (dispatch, getState) => {
    axios({
      method: 'GET',
      url: `${process.env.REACT_APP_API_URL}/patients-details/forUser`,
      headers: {
        Authorization: `Bearer ${Cookies.get('Accesstoken')}`,
      },
    })
      .then(response => {
        dispatch(setPatientsList(response.data));
      })
      .catch(error =>
        setErrorCurrentPatientInfo({
          message: error.message,
          error: error.message,
          status: error.status,
        })
      );
  };
};

export const searchPatientsByText = query => {
  return (dispatch, getState) => {
    axios({
      method: 'GET',
      url: `${process.env.REACT_APP_API_URL}/patients-details/search?search=${query}`,
      headers: {
        Authorization: `Bearer ${Cookies.get('Accesstoken')}`,
      },
    })
      .then(response => {
        dispatch(setPatientsList(response.data));
      })
      .catch(error =>
        dispatch(
          setErrorCurrentPatientInfo({
            message: error.message,
            error: error.message,
            status: error.status,
          })
        )
      );
  };
};

export const fetchPatientById = patientId => {
  return (dispatch, getState) => {
    axios({
      method: 'GET',
      url: `${process.env.REACT_APP_API_URL}/patients-details/${patientId}`,
      headers: {
        Authorization: `Bearer ${Cookies.get('Accesstoken')}`,
      },
    })
      .then(response => {
        dispatch(currentPatientInfoFetched(response.data));
      })
      .catch(error =>
        dispatch(
          setErrorCurrentPatientInfo({
            error: error.message,
            status: error.status,
          })
        )
      );
  };
};

export const createPatientDetails = patient => {
  return (dispatch, getState) => {
    axios({
      method: 'POST',
      url: `${process.env.REACT_APP_API_URL}/patients-details/`,
      headers: {
        Authorization: `Bearer ${Cookies.get('Accesstoken')}`,
      },
      data: {
        ...patient,
      },
    })
      .then(response => {
        dispatch(fetchPatientById(response.data.id));
        dispatch(
          setErrorCurrentPatientInfo({
            message: response.data.successMessage,
            error: null,
            status: response.status,
          })
        );
        dispatch(
          setErrorCurrentPatientInfo({
            message: null,
            error: null,
            status: null,
          })
        );
      })
      .catch(error => {
        dispatch(
          setErrorCurrentPatientInfo({
            error: error.response.data,
            status: error.status,
          })
        );
      });
  };
};

export const updatePatientDetails = patient => {
  return (dispatch, getState) => {
    axios({
      method: 'PATCH',
      url: `${process.env.REACT_APP_API_URL}/patients-details/`,
      headers: {
        Authorization: `Bearer ${Cookies.get('Accesstoken')}`,
      },
      data: {
        ...patient,
      },
    })
      .then(response => {
        dispatch(
          setErrorCurrentPatientInfo({
            message: response.data.successMessage,
            error: null,
            status: response.status,
          })
        );
        dispatch(
          setErrorCurrentPatientInfo({
            message: null,
            error: null,
            status: null,
          })
        );
        dispatch(fetchPatientById(patient.id));
      })
      .catch(error =>
        dispatch(
          setErrorCurrentPatientInfo({
            error: error.response.data,
            status: error.status,
          })
        )
      );
  };
};

export const removePatientDetailsById = id => {
  return (dispatch, getState) => {
    axios({
      method: 'DELETE',
      url: `${process.env.REACT_APP_API_URL}/patients-details/delete`,
      headers: {
        Authorization: `Bearer ${Cookies.get('Accesstoken')}`,
      },
      data: { id },
    })
      .then(response => {
        dispatch(
          setErrorCurrentPatientInfo({
            message: response.data.successMessage,
            error: null,
            status: response.status,
          })
        );
        dispatch(
          setErrorCurrentPatientInfo({
            message: null,
            error: null,
            status: null,
          })
        );
        dispatch(getAllPatientsForAdmin());
      })
      .catch(error =>
        dispatch(
          setErrorCurrentPatientInfo({
            error: error.response.data,
            status: error.status,
          })
        )
      );
  };
};

export const setPatientsList = data => ({
  type: PATIENT_LIST_FETCHED,
  payload: data,
});

export const currentPatientInfoFetched = data => ({
  type: PATIENT_INFO_FETCHED,
  payload: data,
});

export const setErrorCurrentPatientInfo = data => ({
  type: PATIENT_INFO_FETCHED_ERROR,
  payload: data,
});
