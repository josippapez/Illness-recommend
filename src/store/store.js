import { createStore, applyMiddleware } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

import { reducers } from './reducers';

const middleware = [];

const persistConfig = {
  key: 'root',
  storage,
};
const persistedReducer = persistReducer(persistConfig, reducers);

middleware.push(thunk);
if (process.env.NODE_ENV === 'development') middleware.push(logger);

const store = createStore(persistedReducer, applyMiddleware(...middleware));

const persistor = persistStore(store);

export { persistor, store };
