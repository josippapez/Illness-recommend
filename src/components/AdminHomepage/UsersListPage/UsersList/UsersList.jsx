import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Container, Table } from 'reactstrap';
import { useDispatch, useSelector } from 'react-redux';
import ListHeader from '../../../SharedComponents/ListHeader/ListHeader';
import '../../../../styles/ListTable.scss';
import { userInfoFetched } from '../../../../store/actions/usersActions';

const UsersList = props => {
  const currentUser = useSelector(state => state.user);
  const dispatch = useDispatch();
  const [headers, setHeaders] = useState([
    {
      header: 'Ime i prezime',
      headerKey: 'name',
    },
    {
      header: 'Email',
      headerKey: 'email',
    },
    {
      header: 'Rola',
      headerKey: 'role',
    },
    {
      header: 'Akcije',
    },
  ]);
  return (
    <table style={{ width: '100%' }} className="list-table">
      <ListHeader setHeaders={setHeaders} headers={headers} />
      <tbody className="list-table__item-row">
        {props.usersList.map(user => {
          return (
            user.id !== currentUser.id && (
              <tr className="spacer  item-row" key={user.id}>
                <td>{user.name}</td>
                <td>{user.email}</td>
                <td>{user.role}</td>
                <td>
                  <button
                    id="link-to-student-page"
                    onClick={() => {
                      dispatch(userInfoFetched({ data: null }));
                      props.setUserId(user.id);
                      props.setShowUserInfoModal(true);
                    }}
                  >
                    Odaberi
                  </button>
                </td>
              </tr>
            )
          );
        })}
      </tbody>
    </table>
  );
};

UsersList.propTypes = {
  usersList: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  setUserId: PropTypes.func,
  setShowUserInfoModal: PropTypes.func,
};

export default UsersList;
