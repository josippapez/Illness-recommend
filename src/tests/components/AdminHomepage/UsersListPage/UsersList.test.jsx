/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';

import UsersList from '../../../../components/AdminHomepage/UsersListPage/UsersList/UsersList';
import { userInfoFetched } from '../../../../store/actions';

jest.mock('../../../../store/actions');

const props = {
  usersList: [
    {
      id: 1,
      email: 'test@tt.com',
      name: 'Anno Anno',
      role: 'user',
    },
    {
      id: 5,
      email: 'a@tt.com',
      name: 'Crona-Mante',
      role: 'user',
    },
  ],
  setUserId: jest.fn(),
  setShowUserInfoModal: jest.fn(),
};

const mockStore = configureStore();
const store = mockStore({
  user: {
    id: 4,
    name: 'Josip PapeÅ¾',
    email: 'tt@tt.com',
    role: 'admin',
    isAuthenticated: true,
    error: null,
    status: null,
    message: null,
    userInfo: {
      data: {
        id: null,
        name: null,
        email: null,
        role: null,
      },
      error: null,
      status: null,
    },
  },
});

describe('Users list', () => {
  let component;
  beforeEach(() => {
    userInfoFetched.mockImplementation(() => ({ type: 'test' }));
    component = mount(
      <Provider store={store}>
        <UsersList {...props} />
      </Provider>
    );
  });

  it('renders correctly', () => {
    expect(component.html()).toMatchSnapshot();
  });

  it('modal opens on user select button', () => {
    component.find('#link-to-medication-page').first().simulate('click');
    expect(userInfoFetched).toHaveBeenCalled();
  });
});
