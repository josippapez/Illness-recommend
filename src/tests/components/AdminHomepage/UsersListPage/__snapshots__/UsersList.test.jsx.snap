// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users list renders correctly 1`] = `"<table style=\\"width: 100%;\\" class=\\"list-table\\"><thead class=\\"list-table__table-header\\"><tr class=\\"list-table__table-header__row\\"><th class=\\"list-table__table-header__row__header\\" id=\\"name\\">Ime i prezime<span></span></th><th class=\\"list-table__table-header__row__header\\" id=\\"email\\">Email<span></span></th><th class=\\"list-table__table-header__row__header\\" id=\\"role\\">Rola<span></span></th><th class=\\"list-table__table-header__row__header\\">Akcije<span></span></th></tr></thead><tbody class=\\"list-table__item-row\\"><tr class=\\"spacer  item-row\\"><td>Anno Anno</td><td>test@tt.com</td><td>user</td><td style=\\"width: 200px; min-width: 50px;\\"><button id=\\"link-to-medication-page\\">Odaberi</button></td></tr><tr class=\\"spacer  item-row\\"><td>Crona-Mante</td><td>a@tt.com</td><td>user</td><td style=\\"width: 200px; min-width: 50px;\\"><button id=\\"link-to-medication-page\\">Odaberi</button></td></tr></tbody></table>"`;
